session_0/word_processing.rb

```diff
@@ -34,11 +55,35 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  i = 0
+  while ((word_1[i] == word_2[i]) && (i < word_1.length) && (i < word_2.length)) do
+    i += 1
+  end
+  return i
 end
 
 # Given a chosen word and an array of words, return an array of word(s)
 # with the maximum similarity score in the order they appear.
 def most_similar_words(chosen_word, words)
-  raise NotImplementedError # TODO
+  i = 0
+  l = words.length
+  output = Array.new(0)
+  max_sim = 0
+  while i < l do
```

Comment:
Check out the comments ive made above and see if you  could use any of those here.

All the best,
 feel free to contact me @addhyanmalhotra if you have any doubts :) 
 (PS ik painful to move over to ruby after using a traditional language 😂😅)